src/stream/src/executor/source/mod.rs

```diff
@@ -42,3 +49,51 @@ pub async fn barrier_to_message_stream(mut rx: UnboundedReceiver<Barrier>) {
     }
     bail!("barrier reader closed unexpectedly");
 }
+
+pub fn get_split_offset_mapping_from_chunk(
```

Comment:
It seems we still rely on the `split_offset_mapping` to update source internal state, do you plan to refactor that part in future? We should ensure the semantic is same as before I think.

Reply:
`split_offset_mapping` is actually the latest offset of each partitions, we cannot avoid this round of iteration to compute it.